genesyscloud/flow_loglevel/resource_genesyscloud_flow_loglevel_utils.go

```diff
@@ -0,0 +1,83 @@
+package flow_loglevel
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v125/platformclientv2"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+)
+
+func generateFlowLogLevelResource(
+	flowId string,
+	flowLoglevel string,
+	resourceId string,
+) string {
+	return fmt.Sprintf(`resource "genesyscloud_flow_loglevel" "%s" {
+	  flow_id					= "%s"
+	  flow_log_level 			= "%s"
+	}`,
+		resourceId,
+		flowId,
+		flowLoglevel)
+}
+
+// getFlowLogLevelSettingsRequestFromResourceData maps data from schema ResourceData object to a platformclientv2.Flowloglevelrequest
+func getFlowLogLevelSettingsRequestFromResourceData(d *schema.ResourceData) platformclientv2.Flowloglevelrequest {
+	return platformclientv2.Flowloglevelrequest{
+		LogLevelCharacteristics: getFlowLogLevelFromResourceData(d),
+	}
+}
+
+// getFlowLogLevelFromResourceData maps data from schema ResourceData object to a platformclientv2.Flowloglevel
+func getFlowLogLevelFromResourceData(d *schema.ResourceData) *platformclientv2.Flowloglevel {
+	level := d.Get("flow_log_level").(string)
+	if len(d.Get("flow_characteristics").([]interface{})) > 0 {
+		return &platformclientv2.Flowloglevel{
+			Level:           &level,
+			Characteristics: getFlowLogLevelCharacteristicsFromResourceData(d),
+		}
+	} else {
+		return &platformclientv2.Flowloglevel{
+			Level: &level,
+		}
+	}
+
+}
+
+// getFlowLogLevelCharacteristicsFromResourceData maps data from schema ResourceData object to a platformclientv2.Flowcharacteristics
+func getFlowLogLevelCharacteristicsFromResourceData(d *schema.ResourceData) *platformclientv2.Flowcharacteristics {
+	characteristics := d.Get("flow_characteristics").([]interface{})[0].(map[string]interface{})
```

Comment:
A panic could be thrown here. Since all the fields in `flow_characteristics` are optional, the user could define the block like this `flow_characteristics{}` , then the list would have one item and it will be nil. 

As a work around, you can just define the map variable like this

```golang
characteristics, ok := d.Get("flow_characteristics").([]interface{})[0].(map[string]interface{})
if !ok {
  return nil
}
```